docs/source/deployment/prefect.md

```diff
@@ -113,8 +92,124 @@ def build_and_register_flow(pipeline_name, env):
+        super().__init__(name=f"{package_name}_initialization", *args, **kwargs)
+
+    def run(self) -> Tuple[DataCatalog, KedroSession]:
+        """Initializes a Kedro session and returns the DataCatalog and KedroSession"""
+        # bootstrap project within task / flow scope
+        bootstrap_project(self.project_path)
+
+        session = KedroSession.create(
+            project_path=self.project_path, env=self.env, extra_params=self.extra_params
+        )
+        # Note that for logging inside a Prefect task self.logger is used.
+        self.logger.info("Session created with ID %s", session.session_id)
+        pipeline = pipelines.get(self.pipeline_name)
+        context = session.load_context()
+        catalog = context.catalog
+
+        unregistered_ds = pipeline.data_sets() - set(catalog.list())  # type: ignore[union-attr]
+        for ds_name in unregistered_ds:
+            catalog.add(ds_name, MemoryDataSet())
+        return catalog, session
+
+
+class KedroTask(Task):
+    """Kedro node as a Prefect task."""
+
+    def __init__(self, node: Node):
+        self._node = node
+        super().__init__(name=node.name, tags=node.tags)
+
+    def run(self, catalog, session):
+        run_node(self._node, catalog, _create_hook_manager(), session.session_id)
+
+
+def instantiate_task(
+    node: Node,
+    tasks: Dict[str, Dict[str, Union[KedroTask, List[KedroTask]]]],
+) -> Tuple[KedroTask, Dict[str, Dict[str, Union[KedroTask, List[KedroTask]]]]]:
+    """
+    Function pulls node task from <tasks> dictionary. If node task not available
+    in <tasks> the function instantiates the tasks and adds it to <tasks>. In this
+    way we avoid duplicate instantiations of the same node task.
+
+    Args:
+        node: Kedro node for which a Prefect task is being created.
+        tasks: dictionary mapping node names to a dictionary containing
+        node tasks and parent node tasks.
+
+    Returns: Prefect task for the passed node and task dictionary.
+
+    """
+    if tasks.get(node._unique_key) is not None:
+        node_task = tasks[node._unique_key]["task"]
+    else:
+        node_task = KedroTask(node)
+        tasks[node._unique_key] = {"task": node_task}
+
+    # return tasks as it is mutated. We want to make this obvious to the user.
+    return node_task, tasks  # type: ignore[return-value]
+
+
+def generate_flow(
+    init_task: KedroInitTask,
+    tasks: Dict[str, Dict[str, Union[KedroTask, List[KedroTask]]]],
+):
+    """
+    Constructs a Prefect flow given a task dictionary. Task dictionary
+    maps Kedro node names to a dictionary containing a node task and its
+    parents.
+
+    Args:
+        init_task: Prefect initialisation tasks. Used to instantiate a Kedro
+        session within the scope of a Prefect flow.
+        tasks: dictionary mapping Kedro node names to a dictionary
+        containing a corresponding node task and its parents.
+
+    Returns: None
+    """
+    catalog, session = init_task
```

Comment:
This is the weird bit that generated kedro_prefect_initialization[0] and kedro_prefect_intialization[1].

Not sure what is the correct thing, but it should be a `KedroInitTask` object instead and a session